File: bubbleSort.py
def bubbleSort(arr: list[int]) :
    
    index = len(arr) - 1
    isSorted = False
    while not isSorted :
        isSorted = True
        for nums in range(0, index):
           if(arr[nums] < arr[nums + 1]) :
               isSorted = False
               arr[nums], arr[nums + 1] = arr[nums+1],arr[nums]
            
    return arr

print(bubbleSort([35,36,2,4,5,9,25]))

File: bubbleSort.js
const bubbleSort = (arr) => {
	
	for(let i = arr.length; i > 0; i--) {
		let noSwaps;
		for(let j = 0; j < i - 1; j++) {
			noSwaps = true;
			if(arr[j] < arr[j+1]) {
				// console.log(arr, arr[j], arr[j+1])
				let temp = arr[j]
				arr[j] = arr[j+1]
				arr[j+1] = temp
				noSwaps= false;
			}
		}
		if(noSwaps) break;
	}
	return arr
}

console.log(bubbleSort([35,37,45,29,8, 9,27,36,13,4,3,19]))
